app/src/lib/git/update-index.ts

```diff
@@ -157,3 +157,21 @@ export async function stageFiles(
     }
   }
 }
+
+/**
+ * Unstage all the given files by removing them from the index.
+ */
+export async function unstageFiles(
+  repository: Repository,
+  files: ReadonlyArray<WorkingDirectoryFileChange>
+): Promise<void> {
+  for (const file of files) {
+    const paths = []
+    paths.push(file.path)
+    if (file.oldPath) {
+      paths.push(file.oldPath)
+    }
+
+    await updateIndex(repository, paths, { forceRemove: true })
```

Comment:
I like using `update-index` here as we're already using this elsewhere, but it requires multiple invocations (no way to recurse or pass in a wildcard expression). `git rm --cached -r .` is a neat shortcut and handles recursion, but for most people the initial commit will be just a few files, so it's not a huge win.  @desktop/core any strong feels on this?

Reply:
Are there downsides to the `git rm --cached -r .` approach? I think we should generally steer clear of O(n) operations if at all possible.

Reply:
Not really, the `--cached` bit is important to ensure you don't touch files in the working directory, but aside from that I'm not aware of any downsides.

Reply:
It's literally what `git status` suggests, so I'm :thumbsup: to keeping this to one invocation of Git.